app/modules/intelligence/agents/custom_agents/custom_agent_controller.py

```
@@ -0,0 +1,111 @@
+        self.service = CustomAgentService(db)
+
+    async def create_agent(self, user_id: str, agent_data: AgentCreate) -> Agent:
+        """Create a new custom agent"""
+        try:
+            return await self.service.create_agent(user_id, agent_data)
+        except Exception as e:
+            logger.error(f"Error creating custom agent: {str(e)}")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f"Failed to create custom agent: {str(e)}",
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)